### PR TITLE
[FIX] mrp: Avoid creating new line when changing qty to produce

### PR DESCRIPTION
**Pull request by jbm-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:
When we change quantity to produce in MRP, all component lines are
removed and new line with the updated amount to product are created.
The problem appears when we change the quantity to produce and directly
click on a component line, because we open the line that will be
deleted. So if we change the quantity for this component manually,
this suppressed line will be added with the new value. And the computed
line will be also be added. So the line for one component will be
present twice (one with computed value and one with manual entry value).

Desired behavior after PR is merged:
To avoid this problem, when we change the quantity to produce, we update
information of each line (instead of deleting and creating new one).
So if we click on a component line just after changing the quantity,
we open the updated line (and not the deleted one).

id=2007238

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/views/form/form_controller.js [hunk 1]
```diff
@@ -681,29 +681,33 @@ var FormController = BasicController.extend({
...
-            parentID: data.parentID,
-            readonly: data.readonly,
-            deletable: record ? data.deletable : false,
-            recordID: record && record.id,
-            res_id: record && record.res_id,
-            res_model: data.field.relation,
-            shouldSaveLocally: true,
-            title: (record ? _t("Open: ") : _t("Create ")) + (ev.target.string || data.field.string),
-        }).open();
+        this.model.mutex.getUnlockedDef()
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/form/form_controller.js]: maybe add a one line comment to explain that you sync with the mutex to wait for potential onchanges

File: addons/web/static/tests/views/form_tests.js [hunk 2]
```diff
@@ -3257,6 +3257,96 @@ QUnit.module('Views', {
         form.destroy();
     });
 
+    QUnit.test('properly apply onchange on one2many fields direct click', async function (assert) {
+        var done = assert.async();
+        assert.expect(2);
+
+        var def = testUtils.makeTestPromise();
+
+        // This widged will resolve de deferrd when the dialog opens.
```
Comment (aab-odoo) [inline addons/web/static/tests/views/form_tests.js]: de deferrd?

Comment (aab-odoo) [reply addons/web/static/tests/views/form_tests.js]: widged?

File: addons/web/static/tests/views/form_tests.js [hunk 3]
```diff
@@ -3257,6 +3257,96 @@ QUnit.module('Views', {
...
+                    '</group>' +
+                    '<field name="p">' +
+                        '<tree>' +
+                            '<field name="display_name"/>' +
+                            '<field name="int_field"/>' +
+                        '</tree>' +
+                    '</field>' +
+                '</form>',
+            res_id: 1,
+            debug: true,
```
Comment (aab-odoo) [inline addons/web/static/tests/views/form_tests.js]: remove this

**Comment by sle-odoo** [comment]
@jbm-odoo can you rebase on saas-12.5? there are some unrelated commits due to your change of target

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**